base/log.hpp

```diff
@@ -72,4 +55,57 @@ void log_msg(const std::string& msg);
 // TODO(legend): may port more functions here from glog.
 
 }  // namespace base
+
+/// Because of including the gperftools/malloc_extension.h of TCMALLOC,
+/// there is a conflict between the namespace ::base which is belonging to gperftools
+/// and the namaspace husky::base. In order to slove this problem, we use 2 warpper classes
+/// "HuskyLogger_Warpper" and "HuskyLoggerVoidify_Warpper" in namespace husky.
```

Comment:
Let me try to explain. TCMalloc also has a top-level namespace "base", which creates the conflict. In order to re-produce the problem, you may directly include TCMalloc headers using a stable Husky, and it won't build. @MiaoLoud Am I right?